app/code/Magento/Directory/Model/Currency.php

```diff
@@ -488,7 +488,7 @@ private function setOptions(array $options): void
             $this->numberFormatter->setSymbol(\NumberFormatter::CURRENCY_SYMBOL, '');
         }
         if (array_key_exists('precision', $options)) {
-            $this->numberFormatter->setAttribute(\NumberFormatter::FRACTION_DIGITS, $options['precision']);
+            $this->numberFormatter->setAttribute(\NumberFormatter::FRACTION_DIGITS, $options['precision'] ?? 0);
```

Comment:
I don't think thats right place for fix. There are few places that use `null` as a value for precision while default is 2 \Magento\Framework\Pricing\PriceCurrencyInterface::DEFAULT_PRECISION
https://github.com/magento/magento2/blob/548ac72cab788d7ad09393cb838cce39d399061d/lib/internal/Magento/Framework/Pricing/PriceCurrencyInterface.php#L50-L56

I would ask you to test and see if we really expect\want `0` as precision in 
https://github.com/magento/partners-magento2b2b/blob/32b75f9bc94fb4ddf9aa3a07393179554406eab8/app/code/Magento/PurchaseOrder/Ui/Component/Listing/Column/Price.php#L81-L86

https://github.com/magento/partners-magento2b2b/blob/32b75f9bc94fb4ddf9aa3a07393179554406eab8/app/code/Magento/CompanyCredit/Controller/Adminhtml/Index/Reimburse.php#L146-L152

If `0` is desired value please replace `null` with `0`